### PR TITLE
Feature/swed 2147 button component updates

### PR DESCRIPTION
**Pull request by eskilsen**
## Description

New design for buttons! 
Updated Primary and secondary and Icon buttons
Added Tertiary button
Removed all delete-buttons

## Motivation and Context

Part of creating a new design for our system
## How Has This Been Tested?

Ran all tests. Some fails due to React 18. This should have been resolved in another branch

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (this will happen in another branch)
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have updated the **CHANGELOG** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

## Review instructions
[Review instructions](../REVIEW_INSTRUCTIONS.md)

### CODE REVIEW

File: src/less/components/topbar.less
Comment (goldenraphti) [inline src/less/components/topbar.less:1]: - search icon btn needs .btn-icon so it's not black icon over black BG

File: src/less/components/topbar.less [hunk 1]
```diff
@@ -328,7 +328,7 @@
             background-color: transparent;
 
             /* padding-top-bottom, padding-left-right, font-size, line-height */
-            .button-size(0.5rem; 0.5rem; 1rem; 1rem; @btn-border-radius-base;);
+            //.button-size(0.5rem; 0.5rem; 1rem; 1rem; @btn-border-radius-base;); IS THIS NEEDED?
```
Comment (goldenraphti) [inline src/less/components/topbar.less:331]: if not needed then let's remove it
(tested & compared visually `.topbar-close` and it seems fine, identical to main branch)

File: src/less/components/payex/button.less [hunk 2]
```diff
@@ -1,6 +1,6 @@
 @import "../button.less";
 
-&.btn {
+&.new-btn {
```
Comment (goldenraphti) [inline src/less/components/payex/button.less:3]: revert back to `.btn`

File: src/less/variables-payex.less [hunk 3]
```diff
@@ -143,3 +143,47 @@ body {
...
+    --button-bg-disabled: var(--brand-secondary-light-2);
+    --brand-secondary: #3c3c3c;
+    --brand-secondary-hover: #5f5f5f;
+    --brand-orange: #f35b1c;
+
+    @btn-secondary-color:                       @white;
+    @btn-secondary-bg:                          @brand-secondary;
+    @btn-secondary-hover:                       @brand-primary;
+    @btn-secondary-disabled-color:              @white;
+    @btn-secondary-disabled-bg:                 @brand-secondary-light-2;
```
Comment (goldenraphti) [inline src/less/variables-payex.less:168]: remove

File: src/less/variables-payex.less [hunk 4]
```diff
@@ -143,3 +143,47 @@ body {
...
+    --btn-secondary-hover: var(--brand-secondary);
+    // btn-tertiary
+    --btn-tertiary-color: var(--brand-secondary);
+    --btn-tertiary-text-color: var(--white);
+    --btn-tertiary-hover: var(--brand-secondary-hover);
+    --btn-tertiary-disabled: var(--btn-bg-disabled);
+    // Shared button variables
+    --btn-disabled-text: var(--text-disabled);
+    --btn-bg-disabled: rgba(212, 196, 188, 0.3);
+    --btn-active-color: var(--brand-orange);
```
Comment (goldenraphti) [inline src/less/variables-payex.less:188]: orange looks very "destructive" to me 🤔 , could we use another colour maybe?

File: src/less/components/button.less [hunk 5]
```diff
@@ -285,119 +11,73 @@
...
-
-    &.btn-icon-lg {
-        .button-icon-size(0.3rem; 2.5rem; 2.5rem; @font-size-small; @line-height-small; @btn-border-radius-small);
-
-        .material-icons-outlined {
-            font-size: 1.75rem;
+    height: 60px;
+    border-radius: var(--border-radius);
+    font-size: 18px;
+    padding: 12px 16px 12px 16px;
```
Comment (goldenraphti) [inline src/less/components/button.less:21]: Codacy 🤖  is right 👇 

this is enough:
```suggestion
    padding: 12px 16px;
```

File: src/less/components/button.less [hunk 6]
```diff
@@ -1,279 +1,5 @@
...
-@btn-min-width:                             112px;
-
-@btn-font-size-sm:                          1.6rem;
-@btn-font-size-xs:                          1.2rem;
-
-@btn-padding-horizontal:                    1rem;
-@btn-default-height:                        3rem;
-@btn-default-width:                         auto;
-
-.button-group {
```
Comment (goldenraphti) [inline src/less/components/button.less:271]: we're removing `.btn-group` ?
I'm not sure what was its point, but I just want to double-check this is on purpose 😉

File: src/less/components/button.less [hunk 7]
```diff
@@ -424,78 +104,81 @@
...
-    // Specificity overrides
-    &input[type="submit"],
-    &input[type="reset"],
-    &input[type="button"] {
-        &.btn-block {
-            width: 100%;
+        &:not(:last-child) {
+            margin-right: 0.5rem;
         }
     }
```
Comment (goldenraphti) [inline src/less/components/button.less:180]: Bring back to life? 🧙‍♀️ 🪄 🌟

File: src/less/components/button.less [hunk 8]
```diff
@@ -285,119 +11,73 @@
...
-
-        &:not(:last-child) {
-            margin-right: 0.5rem;
-        }
-    }
-
-    &.btn-icon-xl {
-        .button-icon-size(0.45rem; 3rem; 3rem; @font-size-small; @line-height-small; @btn-border-radius-small);
-
-        .material-icons-outlined {
```
Comment (goldenraphti) [inline src/less/components/button.less:312]: `.material-icons-outlined` "font-size" declarations have been deleted. I am not sure it is supposed to be deleted (but I had previously no awareness we had specific font-size declarations for outlined)

File: src/App/ComponentsDocumentation/components/Buttons/constants.js [hunk 9]
```diff
@@ -83,6 +88,37 @@ const ButtonOptions = idAddOn => (
     }
 );
 
+const IconButtonOptions = idAddOn => (
```
Comment (goldenraphti) [inline src/App/ComponentsDocumentation/components/Buttons/constants.js:91]: we're now only offering the size options for the icon buttons, not the Primary/secondary etc anymore,
is this on purpose?
bring it back I guess 😉

File: src/less/components/button.less [hunk 10]
```diff
@@ -285,119 +11,79 @@
...
+        padding: 12px 16px;
+
+        &.btn-icon {
+            i {
+                font-size: 24px;
+            }
         }
     }
 
-    &.btn-icon-xs {
```
[the remark this answers is not in the document]
Comment (eskilsen) [reply src/less/components/button.less:333]: I agree, that said, I really don't think anyone is using the icon-size-classes.. They were fairly new and we have not "announced" in any way.. Also, wouldn't this release be a breaking either way? 
IMO it is better the new way now, that you have one class for styling the button and one for deciding the sizes :D

File: src/App/ComponentsDocumentation/components/Buttons/constants.js (hunk 9 above)
Comment (eskilsen) [reply src/App/ComponentsDocumentation/components/Buttons/constants.js:91]: What do you mean? 😅 It works for me?

**Review by goldenraphti — APPROVED** [review]

**Review by goldenraphti — APPROVED** [review]

**Pull request merged**